calico_kubernetes/calico_kubernetes.py

```diff
@@ -146,35 +143,29 @@ def _configure_profile(self, endpoint):
 
         Currently assumes one pod with each name.
         """
-        pod = self._get_pod_config()
-
         logger.info('Configuring Pod Profile: %s', self.profile_name)
 
         if self._datastore_client.profile_exists(self.profile_name):
-            logger.error("Profile with name %s already exists, exiting.",
-                         self.profile_name)
-            sys.exit(1)
+            logger.warning("Profile with name %s already exists.",
+                           self.profile_name)
         else:
             self._datastore_client.create_profile(self.profile_name)
-
-        self._apply_rules(pod)
-
-        self._apply_tags(pod)
+            self._apply_rules()
 
         # Also set the profile for the workload.
         logger.info('Setting profile %s on endpoint %s',
                     self.profile_name, endpoint.endpoint_id)
-        self._datastore_client.set_profiles_on_endpoint(
-            [self.profile_name], endpoint_id=endpoint.endpoint_id
-        )
+
+        self._datastore_client.append_profiles_to_endpoint(profile_names=[self.profile_name],
```

Comment:
The policy agent is not going to apply policy before the plugin does, because it relies on the endpoint ID annotation that the plugin patches on to the pod object.  Right?

I think the plugin should be using SET here to ensure the initial state is what we expect.